### ISSUE TITLE
bash build.sh command fails 

### ISSUE THREAD

**Issue by surabhi96** [facebookresearch/SparseConvNet#162]
running bdist_egg
running egg_info
creating sparseconvnet.egg-info
writing sparseconvnet.egg-info/PKG-INFO
writing dependency_links to sparseconvnet.egg-info/dependency_links.txt
writing top-level names to sparseconvnet.egg-info/top_level.txt
writing manifest file 'sparseconvnet.egg-info/SOURCES.txt'
package init file 'sparseconvnet/SCN/__init__.py' not found (or not a regular file)
reading manifest file 'sparseconvnet.egg-info/SOURCES.txt'
writing manifest file 'sparseconvnet.egg-info/SOURCES.txt'
installing library code to build/bdist.linux-x86_64/egg
running install_lib
running build_py
creating build
creating build/lib.linux-x86_64-3.7
creating build/lib.linux-x86_64-3.7/sparseconvnet
copying sparseconvnet/submanifoldConvolution.py -> build/lib.linux-x86_64-3.7/sparseconvnet
copying sparseconvnet/activations.py -> build/lib.linux-x86_64-3.7/sparseconvnet
copying sparseconvnet/utils.py -> build/lib.linux-x86_64-3.7/sparseconvnet
copying sparseconvnet/randomizedStrideMaxPooling.py -> build/lib.linux-x86_64-3.7/sparseconvnet
copying sparseconvnet/metadata.py -> build/lib.linux-x86_64-3.7/sparseconvnet
copying sparseconvnet/tables.py -> build/lib.linux-x86_64-3.7/sparseconvnet
copying sparseconvnet/permutohedralSubmanifoldConvolution.py -> build/lib.linux-x86_64-3.7/sparseconvnet
copying sparseconvnet/deconvolution.py -> build/lib.linux-x86_64-3.7/sparseconvnet
copying sparseconvnet/denseToSparse.py -> build/lib.linux-x86_64-3.7/sparseconvnet
copying sparseconvnet/inputBatch.py -> build/lib.linux-x86_64-3.7/sparseconvnet
copying sparseconvnet/unPooling.py -> build/lib.linux-x86_64-3.7/sparseconvnet
copying sparseconvnet/averagePooling.py -> build/lib.linux-x86_64-3.7/sparseconvnet
copying sparseconvnet/fullConvolution.py -> build/lib.linux-x86_64-3.7/sparseconvnet
copying sparseconvnet/sparseToDense.py -> build/lib.linux-x86_64-3.7/sparseconvnet
copying sparseconvnet/networkArchitectures.py -> build/lib.linux-x86_64-3.7/sparseconvnet
copying sparseconvnet/classificationTrainValidate.py -> build/lib.linux-x86_64-3.7/sparseconvnet
copying sparseconvnet/shapeContext.py -> build/lib.linux-x86_64-3.7/sparseconvnet
copying sparseconvnet/sparseConvNetTensor.py -> build/lib.linux-x86_64-3.7/sparseconvnet
copying sparseconvnet/spectral_norm.py -> build/lib.linux-x86_64-3.7/sparseconvnet
copying sparseconvnet/batchNormalization.py -> build/lib.linux-x86_64-3.7/sparseconvnet
copying sparseconvnet/dropout.py -> build/lib.linux-x86_64-3.7/sparseconvnet
copying sparseconvnet/ioLayers.py -> build/lib.linux-x86_64-3.7/sparseconvnet
copying sparseconvnet/__init__.py -> build/lib.linux-x86_64-3.7/sparseconvnet
copying sparseconvnet/identity.py -> build/lib.linux-x86_64-3.7/sparseconvnet
copying sparseconvnet/networkInNetwork.py -> build/lib.linux-x86_64-3.7/sparseconvnet
copying sparseconvnet/sparsify.py -> build/lib.linux-x86_64-3.7/sparseconvnet
copying sparseconvnet/randomizedStrideConvolution.py -> build/lib.linux-x86_64-3.7/sparseconvnet
copying sparseconvnet/convolution.py -> build/lib.linux-x86_64-3.7/sparseconvnet
copying sparseconvnet/sequential.py -> build/lib.linux-x86_64-3.7/sparseconvnet
copying sparseconvnet/maxPooling.py -> build/lib.linux-x86_64-3.7/sparseconvnet
running build_ext
building 'sparseconvnet.SCN' extension
creating build/temp.linux-x86_64-3.7
creating build/temp.linux-x86_64-3.7/sparseconvnet
creating build/temp.linux-x86_64-3.7/sparseconvnet/SCN
/usr/local/cuda/bin/nvcc -I/home/abhishek/surabhi/SparseConvNet/sparseconvnet/SCN/ -I/home/abhishek/anaconda3/envs/pointpillars/lib/python3.7/site-packages/torch/include -I/home/abhishek/anaconda3/envs/pointpillars/lib/python3.7/site-packages/torch/include/torch/csrc/api/include -I/home/abhishek/anaconda3/envs/pointpillars/lib/python3.7/site-packages/torch/include/TH -I/home/abhishek/anaconda3/envs/pointpillars/lib/python3.7/site-packages/torch/include/THC -I/usr/local/cuda/include -I/home/abhishek/anaconda3/envs/pointpillars/include/python3.7m -c sparseconvnet/SCN/cuda.cu -o build/temp.linux-x86_64-3.7/sparseconvnet/SCN/cuda.o -D__CUDA_NO_HALF_OPERATORS__ -D__CUDA_NO_HALF_CONVERSIONS__ -D__CUDA_NO_HALF2_OPERATORS__ --expt-relaxed-constexpr --compiler-options '-fPIC' -std=c++14 -Xcompiler -fopenmp -DTORCH_API_INCLUDE_EXTENSION_H -DTORCH_EXTENSION_NAME=SCN -D_GLIBCXX_USE_CXX11_ABI=0 -gencode=arch=compute_75,code=sm_75
/home/abhishek/anaconda3/envs/pointpillars/lib/python3.7/site-packages/torch/include/c10/util/ArrayRef.h(278): warning: attribute does not apply to any entity

/home/abhishek/anaconda3/envs/pointpillars/lib/python3.7/site-packages/torch/include/c10/core/TensorTypeSet.h(44): warning: integer conversion resulted in a change of sign

/home/abhishek/anaconda3/envs/pointpillars/lib/python3.7/site-packages/torch/include/c10/util/ArrayRef.h(278): warning: attribute does not apply to any entity

/home/abhishek/anaconda3/envs/pointpillars/lib/python3.7/site-packages/torch/include/c10/core/TensorTypeSet.h(44): warning: integer conversion resulted in a change of sign

/usr/include/c++/7/bits/basic_string.tcc: In instantiation of ‘static std::basic_string<_CharT, _Traits, _Alloc>::_Rep* std::basic_string<_CharT, _Traits, _Alloc>::_Rep::_S_create(std::basic_string<_CharT, _Traits, _Alloc>::size_type, std::basic_string<_CharT, _Traits, _Alloc>::size_type, const _Alloc&) [with _CharT = char16_t; _Traits = std::char_traits<char16_t>; _Alloc = std::allocator<char16_t>; std::basic_string<_CharT, _Traits, _Alloc>::size_type = long unsigned int]’:
/usr/include/c++/7/bits/basic_string.tcc:578:28:   required from ‘static _CharT* std::basic_string<_CharT, _Traits, _Alloc>::_S_construct(_InIterator, _InIterator, const _Alloc&, std::forward_iterator_tag) [with _FwdIterator = const char16_t*; _CharT = char16_t; _Traits = std::char_traits<char16_t>; _Alloc = std::allocator<char16_t>]’
/usr/include/c++/7/bits/basic_string.h:5042:20:   required from ‘static _CharT* std::basic_string<_CharT, _Traits, _Alloc>::_S_construct_aux(_InIterator, _InIterator, const _Alloc&, std::__false_type) [with _InIterator = const char16_t*; _CharT = char16_t; _Traits = std::char_traits<char16_t>; _Alloc = std::allocator<char16_t>]’
/usr/include/c++/7/bits/basic_string.h:5063:24:   required from ‘static _CharT* std::basic_string<_CharT, _Traits, _Alloc>::_S_construct(_InIterator, _InIterator, const _Alloc&) [with _InIterator = const char16_t*; _CharT = char16_t; _Traits = std::char_traits<char16_t>; _Alloc = std::allocator<char16_t>]’
/usr/include/c++/7/bits/basic_string.tcc:656:134:   required from ‘std::basic_string<_CharT, _Traits, _Alloc>::basic_string(const _CharT*, std::basic_string<_CharT, _Traits, _Alloc>::size_type, const _Alloc&) [with _CharT = char16_t; _Traits = std::char_traits<char16_t>; _Alloc = std::allocator<char16_t>; std::basic_string<_CharT, _Traits, _Alloc>::size_type = long unsigned int]’
/usr/include/c++/7/bits/basic_string.h:6688:95:   required from here
/usr/include/c++/7/bits/basic_string.tcc:1067:16: error: cannot call member function ‘void std::basic_string<_CharT, _Traits, _Alloc>::_Rep::_M_set_sharable() [with _CharT = char16_t; _Traits = std::char_traits<char16_t>; _Alloc = std::allocator<char16_t>]’ without object
       __p->_M_set_sharable();
       ~~~~~~~~~^~
/usr/include/c++/7/bits/basic_string.tcc: In instantiation of ‘static std::basic_string<_CharT, _Traits, _Alloc>::_Rep* std::basic_string<_CharT, _Traits, _Alloc>::_Rep::_S_create(std::basic_string<_CharT, _Traits, _Alloc>::size_type, std::basic_string<_CharT, _Traits, _Alloc>::size_type, const _Alloc&) [with _CharT = char32_t; _Traits = std::char_traits<char32_t>; _Alloc = std::allocator<char32_t>; std::basic_string<_CharT, _Traits, _Alloc>::size_type = long unsigned int]’:
/usr/include/c++/7/bits/basic_string.tcc:578:28:   required from ‘static _CharT* std::basic_string<_CharT, _Traits, _Alloc>::_S_construct(_InIterator, _InIterator, const _Alloc&, std::forward_iterator_tag) [with _FwdIterator = const char32_t*; _CharT = char32_t; _Traits = std::char_traits<char32_t>; _Alloc = std::allocator<char32_t>]’
/usr/include/c++/7/bits/basic_string.h:5042:20:   required from ‘static _CharT* std::basic_string<_CharT, _Traits, _Alloc>::_S_construct_aux(_InIterator, _InIterator, const _Alloc&, std::__false_type) [with _InIterator = const char32_t*; _CharT = char32_t; _Traits = std::char_traits<char32_t>; _Alloc = std::allocator<char32_t>]’
/usr/include/c++/7/bits/basic_string.h:5063:24:   required from ‘static _CharT* std::basic_string<_CharT, _Traits, _Alloc>::_S_construct(_InIterator, _InIterator, const _Alloc&) [with _InIterator = const char32_t*; _CharT = char32_t; _Traits = std::char_traits<char32_t>; _Alloc = std::allocator<char32_t>]’
/usr/include/c++/7/bits/basic_string.tcc:656:134:   required from ‘std::basic_string<_CharT, _Traits, _Alloc>::basic_string(const _CharT*, std::basic_string<_CharT, _Traits, _Alloc>::size_type, const _Alloc&) [with _CharT = char32_t; _Traits = std::char_traits<char32_t>; _Alloc = std::allocator<char32_t>; std::basic_string<_CharT, _Traits, _Alloc>::size_type = long unsigned int]’
/usr/include/c++/7/bits/basic_string.h:6693:95:   required from here
/usr/include/c++/7/bits/basic_string.tcc:1067:16: error: cannot call member function ‘void std::basic_string<_CharT, _Traits, _Alloc>::_Rep::_M_set_sharable() [with _CharT = char32_t; _Traits = std::char_traits<char32_t>; _Alloc = std::allocator<char32_t>]’ without object
error: command '/usr/local/cuda/bin/nvcc' failed with exit status 1

Note: 
Cuda version : 10.1
gcc version: 7.5

**Comment by zwqnju** [comment]
I have the same error. I update PyTorch to 1.4 and solve it: `conda install pytorch torchvision cudatoolkit=10.1 -c pytorch`

**Comment by xandox** [comment]
Also helps to use gcc-6
set environment variable for use gcc-6
```
export CC=/usr/bin/gcc-6
export CXX=/usr/bin/g++-6
```
in setup.py for nvcc arguments add `-ccbin /usr/bin/g++-6`

was
```
extra = {'cxx': ['-std=c++14', '-fopenmp'], 'nvcc': ['-std=c++14', '-Xcompiler', '-fopenmp']}
```

now
```
extra = {'cxx': ['-std=c++14', '-fopenmp'], 'nvcc': ['-std=c++14', '-Xcompiler', '-fopenmp', '-ccbin', '/usr/bin/g++=6']}
```

may be there is some easier way to setup concrete compiler for use but I do not know it. If some one show one I will be thankful.

**Comment by resplendent-star** [comment]
> 还有助于使用gcc-6
> 设置环境变量以使用gcc-6
> 
> ```
> export CC=/usr/bin/gcc-6
> export CXX=/usr/bin/g++-6
> ```
> 
> 在setup.py中为nvcc参数添加 `-ccbin /usr/bin/g++-6`
> 
> 原为
> 
> ```
> extra = {'cxx': ['-std=c++14', '-fopenmp'], 'nvcc': ['-std=c++14', '-Xcompiler', '-fopenmp']}
> ```
> 
> 现在
> 
> ```
> extra = {'cxx': ['-std=c++14', '-fopenmp'], 'nvcc': ['-std=c++14', '-Xcompiler', '-fopenmp', '-ccbin', '/usr/bin/g++=6']}
> ```
> 
> 可能有一些更简单的方法来设置要使用的具体编译器，但我不知道。如果有人表演一个，我将很感激。

hi,i am work with this method,thank you very much!!!!!!

**Comment by llllliu1** [comment]
hey，i have the same problem，did u solve it？
（ptorch1.4，cuda10.1，gcc4.8.5）

**Comment by llllliu1** [comment]
> > 还有助于使用gcc-6
> > 设置环境变量以使用gcc-6
> > ```
> > export CC=/usr/bin/gcc-6
> > export CXX=/usr/bin/g++-6
> > ```
> > 
> > 
> > 在setup.py中为nvcc参数添加 `-ccbin /usr/bin/g++-6`
> > 原为
> > ```
> > extra = {'cxx': ['-std=c++14', '-fopenmp'], 'nvcc': ['-std=c++14', '-Xcompiler', '-fopenmp']}
> > ```
> > 
> > 
> > 现在
> > ```
> > extra = {'cxx': ['-std=c++14', '-fopenmp'], 'nvcc': ['-std=c++14', '-Xcompiler', '-fopenmp', '-ccbin', '/usr/bin/g++=6']}
> > ```
> > 
> > 
> > 可能有一些更简单的方法来设置要使用的具体编译器，但我不知道。如果有人表演一个，我将很感激。
> 
> hi,i am work with this method,thank you very much!!!!!!



> > 还有助于使用gcc-6
> > 设置环境变量以使用gcc-6
> > ```
> > export CC=/usr/bin/gcc-6
> > export CXX=/usr/bin/g++-6
> > ```
> > 
> > 
> > 在setup.py中为nvcc参数添加 `-ccbin /usr/bin/g++-6`
> > 原为
> > ```
> > extra = {'cxx': ['-std=c++14', '-fopenmp'], 'nvcc': ['-std=c++14', '-Xcompiler', '-fopenmp']}
> > ```
> > 
> > 
> > 现在
> > ```
> > extra = {'cxx': ['-std=c++14', '-fopenmp'], 'nvcc': ['-std=c++14', '-Xcompiler', '-fopenmp', '-ccbin', '/usr/bin/g++=6']}
> > ```
> > 
> > 
> > 可能有一些更简单的方法来设置要使用的具体编译器，但我不知道。如果有人表演一个，我将很感激。
> 
> hi,i am work with this method,thank you very much!!!!!!

Excuse me, is this related to the gcc version? My gccC version is 4.8.5. Do I need to upgrade?
